src/ui/components/JEXLDebuggerPage.tsx

```diff
@@ -0,0 +1,81 @@
+import { ChangeEvent, FC, useState, useCallback, useEffect } from "react";
+
+const JEXLDebuggerPage: FC = () => {
+  const [clientContext, setClientContext] = useState({});
+  const [jexlExpression, setJexlExpression] = useState("");
+  const [output, setOutput] = useState("");
+
+  useEffect(() => {
+    void (async () => {
+      try {
+        const context = await browser.experiments.nimbus.getClientContext();
+        setClientContext(context);
+      } catch (error) {
+        console.error("Error fetching client context:", error);
+      }
+    })();
+  }, []);
+
+  const handleExpressionChange = useCallback(
+    (event: ChangeEvent<HTMLTextAreaElement>) => {
+      setJexlExpression(event.target.value);
+    },
+    [],
+  );
+
+  const handleEvaluateClick = useCallback(async () => {
+    try {
+      const result = await browser.experiments.nimbus.evaluateJEXL(
+        jexlExpression,
+        clientContext,
+      );
+      setOutput(result);
+    } catch (error) {
+      console.error(error);
+      setOutput("Error evaluating expression");
+    }
+  }, [jexlExpression, clientContext]);
+
+  return (
+    <div className="main-content">
+      <div className="jexl-debugger">
+        <div className="jexl-debugger__section">
+          <h2 className="jexl-debugger__section-title">
+            JEXL Filter Expression
+          </h2>
+          <textarea
+            value={jexlExpression}
+            onChange={handleExpressionChange}
+            placeholder="Enter JEXL expression here"
+            className="jexl-debugger__expression-input"
+          />
+          <button
+            onClick={handleEvaluateClick}
+            className="jexl-debugger__evaluate-button"
+          >
+            Evaluate
+          </button>
+        </div>
+        <div className="jexl-debugger__section">
+          <h2 className="jexl-debugger__section-title">Output</h2>
+          <pre>{output}</pre>
+        </div>
+        <div className="jexl-debugger__section">
+          <h2 className="jexl-debugger__section-title">Client Context</h2>
+          {Object.entries(clientContext).map(([key, value]) => (
+            <div key={key}>
+              <p className="jexl-debugger__client-context-title">{key}</p>
+              <textarea
+                className="jexl-debugger__client-context-value"
+                readOnly
+                value={JSON.stringify(value, null, 2)}
+              />
+            </div>
+          ))}
```

Comment:
I'm pretty sure this triggers re-renders even if clientContext does not change. We should `useMemo()` the result of `Object.entries(clientContext)`